docs/source/dev/test.rst

```diff
@@ -0,0 +1,107 @@
+.. highlight:: cpp
+
+Writing a unit test
+==================
+
+After implementing a new functionality in Alpaka, it is recommended to test it. Indeed, having no compile time errors
+when building alpaka with any backend doesn't mean that such functionality is well implemented and behaves as expected.
+
+Unit tests are written and integrated with `Catch2`_ and they are standalone executables located in the ``test/unit`` and
+``test/integ`` paths inside the ``build`` folder. They can be built all at once by setting the ``CMake``
+argument ``-DBUILD_TESTING=ON`` and running the command ``cmake --build .``, and then run with the ``ctest`` command as
+described in the :doc:`Installation section </basic/install>`.
```

Comment:
Consider pointing out that the commands are supposed to be run in the build folder.